### ISSUE TITLE
Confusing CLI behavior with Kubernetes Flags

### ISSUE THREAD

**Issue by NurlashKO** [telepresenceio/telepresence#2073]
After specifying any connection flag(s), for example:
```
$ telepresence connect --request-timeout 100
```
All those flags must be included in the proceeding calls:
```
$ telepresence list
telepresence: error: connector.Connect: Cluster configuration changed, please quit telepresence and reconnect
```
```
telepresence list --request-timeout 100    
xxx                        : ready to intercept (traffic-agent already installed)
yyy                        : ready to intercept (traffic-agent not yet installed)
```

It happens due to this check:
https://github.com/telepresenceio/telepresence/blob/release/v2/pkg/client/connector/userd_k8s/k8s_config.go#L150
```golang
// ContextServiceAndFlagsEqual determines if this instance is equal to the given instance with respect to context,
// server, and flag arguments.
func (kf *Config) ContextServiceAndFlagsEqual(okf *Config) bool {
	return kf != nil && okf != nil &&
		kf.Context == okf.Context &&
		kf.Server == okf.Server &&
		sliceEqual(kf.flagArgs, okf.flagArgs)
} 
```
It makes sense because of the next case:
```
$ telepresence connect --kubeconfig my_kubeconfig.txt

# will use default ~/.kube/config file or $KUBECONFIG
$ telepresence list 
``` 
However, developer experience still suffers because of the lots of flags.

Possible solution to this is pushing those flags that were used in initial `connect` statement to all following CLI calls.
Removing flag check from the function above is optional. 

If it sounds good to you, I would love to try implementing it.

**Comment by NurlashKO** [comment]
Another, and probably even better solution, is pushing those flags that were used in initial `connect` statement to all following cli calls.

**Comment by thallgren** [comment]
I think part of the problem is that `telepresence list`, `telepresence intercept`, and other commands, can be invoked without first doing a `telepresence connect`, in which case the connect is implicit. Passing flags to the command becomes ambiguous if flags from the initial connect are also in effect. Should those flags be combined or overridden? And if it's the latter, then how do I override so that I don't have any flags (because that's also a valid flag combo)?

I think it would be acceptable to make it so that only `telepresence connect` accept kubernetes flags and those flags are in effect until a `telepresence quit`. An implicit connect would always be made _without_ the kubernetes flags. In other words, in order to use kubernetes flags, it's required to do an explicit connect. What do you think?

**Comment by NurlashKO** [comment]
@thallgren Thank you for your comment!
Disallowing `kubectl` flags from mentioned commands sounds perfect to me.
I will work on it this weekend.

I have one more question related to the `list/intercept + connect` behavior.
I am personally not a big fan of it.
The reason being that from their signature it is hard to deduce that new k8s resources might be created, hence easier to execute them in a wrong context by mistake and in general it looks strange to me that something is being deployed without explicit changed resource listing + approval.

Sorry if it was already discussed and it is probably completely different issue.
I am wondering what do you think if we also add a simple:
```
Looks like Telepresence daemon was not launched yet.
Run "telepresence connect"(y/n) :
```
or even more broad prompt(example below) in `telepresence connect`?
```
Current context: <context>
"traffic-manager" Pod was not found in your cluster, proceed with deployment(y/n) : 
```

**Comment by thallgren** [comment]
Fixed in 2.5.0.
